### ISSUE TITLE
Error on `keyof` and removing on fixes in  `jsdoc/type-formatting`

### ISSUE THREAD

**Issue by alexander-akait** [gajus/eslint-plugin-jsdoc#1647]
## Expected behavior

Allow to have `keyof` and do not remove on fix

## Actual behavior

Error and remove `keyof` on fix

## ESLint Config

```js
{
  "jsdoc/type-formatting": [
				"error",
				{
					objectTypeBracketSpacing: " ",
					objectFieldSeparator: "comma-and-linebreak",
					objectFieldSeparatorTrailingPunctuation: true,
					trailingPunctuationMultilineOnly: true,
				},
  ],
}
```

## ESLint sample

```js
/**
 * @template {string} K
 * @typedef {K extends keyof AllCodeGenerationSchemas ? AllCodeGenerationSchemas[K] : EXPECTED_ANY} CodeGenValue
 */
```

## Environment

- Node version: v24.11.1
- ESLint version v9.39.2
- `eslint-plugin-jsdoc` version: v62.5.0
